content/2-templating/5-dom-ref/lit/input-focused.js

```diff
@@ -0,0 +1,16 @@
+import { LitElement, html } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import { ref, createRef } from 'lit/directives/ref.js';
+
+@customElement('input-focused')
+export class InputFocused extends LitElement {
+	inputRef = createRef();
+
+	firstUpdated() {
+		this.inputRef.value.focus();
+	}
```

Comment:
Using constructor: https://lit.dev/docs/v1/components/events/#add-event-listeners-in-the-constructor
Usinf `firstUpdated`: https://lit.dev/docs/v1/components/lifecycle/#firstupdated

The `constructor` syntax looks cleaner to me. What do you think ?

Reply:
The first moment on which `inputRef.value` is available is on `firstUpdated`. It'll be undefined on `constructor` and `connectedCallback`! This basically came from the [docs themselves](https://lit.dev/docs/templates/directives/#referencing-rendered-dom).

Reply:
Makes sense !! Thanks for the clarification !